zlib-rs/src/inflate/window.rs

```diff
@@ -20,24 +20,20 @@ pub struct Window<'a> {
 }
 
 impl<'a> Window<'a> {
-    pub fn from_slice(slice: &'a mut [u8]) -> Self {
-        Self {
-            buf: unsafe { slice_to_uninit_mut(slice) },
-            have: 0,
-            next: 0,
-        }
-    }
-
     pub fn into_inner(self) -> &'a mut [MaybeUninit<u8>] {
         self.buf
     }
 
     pub fn is_empty(&self) -> bool {
-        self.buf.len() == 0
+        self.size() == 0
     }
 
     pub fn size(&self) -> usize {
-        self.buf.len()
+        if self.buf.is_empty() {
+            0
+        } else {
+            self.buf.len() - Self::padding()
+        }
```

Comment:
the true cause of these problems: padding should not be part of the window size! An empty buffer is only used in tests or when deallocating the window.